utils_pytorch.py

```diff
@@ -103,33 +105,41 @@ def export_model(model, path=None, input_shape=(1, 3, 64, 64)):
     """
     path = get_model_path() if path is None else path
     model = deepcopy(model).cpu().eval()
-    if not isinstance(model, torch.jit.ScriptModule):
-        assert input_shape is not None, "`input_shape` must be provided since model is not a " \
-                                        "`ScriptModule`."
-        traced_model = trace(model, torch.zeros(*input_shape))
-    else:
-        traced_model = model
-    torch.jit.save(traced_model, path)
-    return path
-
-
-def import_model(path=None):
+    if use_script_module:
+        if not isinstance(model, torch.jit.ScriptModule):
+            assert input_shape is not None, "`input_shape` must be provided since model is not a " \
+                                            "`ScriptModule`."
+            traced_model = trace(model, torch.zeros(*input_shape))
+        else:
+            traced_model = model
+        torch.jit.save(traced_model, path)
+        return path
+    else:
+        torch.save(model, path) # saves model as a nn.Module
+        return path        
+
+
+def import_model(path=None, use_script_module=False):
     """
     Imports a model (as ScriptModule) from file.
 
     Parameters
     ----------
     path : str
         Path to where the model is saved. Defaults to the return value of the `get_model_path`
-        function above.
+    use_script_module : True or False (default = True)
+        If True loads model as torch.jit.ScriptModule. Else loads as a normal Module.
 
     Returns
     -------
-    torch.jit.ScriptModule
+    torch.jit.ScriptModule / torch.nn.Module
         The model file.
     """
     path = get_model_path() if path is None else path
-    return torch.jit.load(path)
+    if use_script_module:
```

Comment:
We must auto-detect If the target file (at `path`) contains a serialised `torch.jit.ScriptModule` or a `torch.nn.Module`. My first hunch would be to try and load the file as if it were a ScriptModule (i.e. with `torch.jit.load`), and if that fails, fall back to `torch.load`.

Reply:
A simple try except error handling can be done if passing the argument is not an option.

try:
       return torch.jit.load(path) # automatic loading error if the path file is not a script module.
except:
       return torch.load(path) # loads the normal module as model

Reply:
This way the code tries to load `torch.jit.ScriptModule` by default using `torch.jit.script`. In case it encounters error due to the `torch.nn.Module`, the error handling should automatically switch to `torch.load`.

I hope this should be fine.

Current problem with torch.jit.ScriptModule is there are lots if constraints on inputs, outputs (like a dictionary is not possible etc.) and some functions like `torch.distributions`.

Handling this would be counterproductive at this point of the challenge I guess.

Reply:
Okay, when you try to load a plain old pickle dump with torch.jit.load, you get back a RuntimeError. You could catch this and fall back to using torch.load. 

Please make sure that import_model does not take any arguments except the path (since this is how it will be used internally).